### PR TITLE
making dataset_git_revision field mandatory

### PR DESCRIPTION
**Pull request by AndreaFrancis**
See https://github.com/huggingface/datasets-server/pull/1988#issuecomment-1766330413 for context
Depends on https://github.com/huggingface/datasets-server/pull/1996

### CODE REVIEW

**Review by severo — APPROVED** [review]

File: libs/libcommon/src/libcommon/utils.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def get_datetime(days: Optional[float] = None) -> datetime:
 
 def inputs_to_string(
     dataset: str,
-    revision: Optional[str] = None,
+    revision: str,
```
[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply libs/libcommon/src/libcommon/utils.py:137]: done

File: services/search/src/search/routes/search.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
 
 import logging
 from http import HTTPStatus
-from pathlib import Path
 from typing import Optional
```
[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply services/search/src/search/routes/search.py]: Not sure, but it was fixed on https://github.com/huggingface/datasets-server/pull/1997

File: services/worker/tests/job_runners/dataset/test_opt_in_out_urls_count.py [hunk 3]
```diff
@@ -197,6 +197,7 @@ def test_compute(
     upsert_response(
         kind="dataset-config-names",
         dataset=dataset,
+        dataset_git_revision="dataset_git_revision",
```
[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply services/worker/tests/job_runners/dataset/test_opt_in_out_urls_count.py]: Done, I added the constant usage in the modified files, but I think we can do same for other tests to standardize some test values (I will try to do it in another PR).

File: services/search/src/search/routes/search.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (severo) [reply services/search/src/search/routes/search.py]: Oh ok, thanks

**Pull request merged**